syntax = "proto3";

package pagekeeper;

service PageKeeper {
  rpc Register (RegisterRequest) returns (RegisterResponse) {}
  rpc Authenticate (AuthenticateRequest) returns (AuthenticateResponse) {}
  rpc Verify (VerifyRequest) returns (VerifyResponse) {}
  rpc FetchUsers (FetchUsersRequest) returns (FetchUsersResponse) {}
}

message User {
  string id = 1;
  string email = 2;
  bool is_admin = 3;
  string last_name = 4;
  string first_name = 5;
}

message RegisterRequest {
  string email = 1;
  string password = 2;
  string first_name = 3;
  string last_name = 4;
  bool is_admin = 5;
}

message RegisterResponse {
  string message = 1;
  string id = 2;
}

message AuthenticateRequest {
  string email = 1;
  string password = 2;
}

message AuthenticateResponse {
  string message = 1;
  string access_token = 2;
  User user = 3;
}

message VerifyRequest {
  string access_token = 1;
}

message VerifyResponse {
  string message = 1;
  User user = 2;
}

message FetchUsersRequest {
  int32 page = 1;
  int32 page_size = 2;
  repeated string ids = 3;
}

message FetchUsersResponse {
  string message = 1;
  int32 total_users = 2;
  int32 current_page = 3;
  repeated User users = 4;
}
